### ISSUE TITLE
LinterStats is not JSON serializable

### ISSUE THREAD

**Issue by TrentDaniel** [Exirel/pylint-json2html#13]
Hi, my company uses pylint for our Python API, and we use your project to present the pylint results in html. Thanks btw, it's very helpful! Once Pylint release it's new version, 12.2.x, on November 25, 2021, we started having problems with presenting the pylint in html using your project. We would consistently get the error 
```
pylint_stderr is: Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/sbox/tdaniel/dev/temporary_checkout/venv/lib/python3.7/site-packages/pylint/lint/run.py", line 399, in __init__
    score_value = linter.generate_reports()
  File "/opt/sbox/tdaniel/dev/temporary_checkout/venv/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1279, in generate_reports
    self.reporter.on_close(self.stats, previous_stats)
  File "/opt/sbox/tdaniel/dev/temporary_checkout/venv/lib/python3.7/site-packages/pylint_json2html/__init__.py", line 249, in on_close
    print(json.dumps(reports, cls=JSONSetEncoder, indent=4), file=self.out)
  File "/home/tdaniel/.pyenv/versions/3.7.0/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/tdaniel/.pyenv/versions/3.7.0/lib/python3.7/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/home/tdaniel/.pyenv/versions/3.7.0/lib/python3.7/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/home/tdaniel/.pyenv/versions/3.7.0/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/home/tdaniel/.pyenv/versions/3.7.0/lib/python3.7/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/opt/sbox/tdaniel/dev/temporary_checkout/venv/lib/python3.7/site-packages/pylint_json2html/__init__.py", line 185, in default
    return super(JSONSetEncoder, self).default(o)
  File "/home/tdaniel/.pyenv/versions/3.7.0/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type LinterStats is not JSON serializable
```
Looking at the file changes for the new version of pylint, it seems that the stats class was changed to LinterStats, and for some reason that is not being parsed by the JSONSetEncoder. The solution to the problem is just to return the dictionary of the LinterStats object from the JSONSetEncoder. I'll open a PR with the changes so you can review them.

**Comment by Exirel** [comment]
Oh. Indeed, a breaking change. How unexpected. 🙃 
Thanks for your interest, I'll be happy to see a PR for that!

**Comment by TrentDaniel** [comment]
Unfortunately, it seems that I can't authenticate while trying to push my commit. So, I'm just going to post the change here, since it is quite small. In pylint_json2html/__init__.py:151, change to the following:
```
class JSONSetEncoder(json.JSONEncoder):
    """Custom JSON Encoder to transform python sets into simple list"""
    def default(self, o):  # pylint: disable=E0202
        if isinstance(o, set):
            return list(o)
        if isinstance(o, LinterStats):
            v = vars(o)
            return v
        return super(JSONSetEncoder, self).default(o)
```
Also, the following needs to be imported at the top of the file:
```
from pylint.utils.linterstats import LinterStats
```

**Comment by TrentDaniel** [comment]
@Exirel, hi, I was wondering if you had had a chance to look at my suggested changes?

**Comment by Exirel** [comment]
I'm looking right now. I'll see what I can do, now that I secured a few hours to work on that. Thanks for your snippet and the time you took to debug the problem, that helps me a lot.

**Comment by Exirel** [comment]
Fixed in latest version 0.4.0. Thanks for your report and your suggestion, really help me getting to the solution way faster.
